src/deque.rs

```diff
@@ -866,4 +1043,55 @@ mod tests {
         q.push_back(0).unwrap();
         assert_eq!(q.len(), 1);
     }
+
+    #[test]
+    fn make_contiguous() {
+        let mut q: Deque<i32, 4> = Deque::new();
+        assert_eq!(q.len(), 0);
+
+        q.push_back(0).unwrap();
+        q.push_back(1).unwrap();
+        q.push_back(2).unwrap();
+        q.push_back(3).unwrap();
+
+        // Deque contains: 0, 1, 2, 3
+        assert_eq!(q.pop_front(), Some(0));
+        assert_eq!(q.pop_front(), Some(1));
+
+        // Deque contains: ., ., 2, 3
+        q.push_back(4).unwrap();
+
+        // Deque contains: 4, ., 2, 3
+        assert_eq!(q.as_slices(), (&[2, 3][..], &[4][..]));
```

Comment:
Can you use e.g. `[2, 3].as_slice()` in the tests? That may be a bit more readable.